src/shared_library.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <string>
+
+#include "rcpputils/shared_library.hpp"
+
+namespace rcpputils
+{
+SharedLibrary::SharedLibrary(const std::string & library_path)
+{
+  lib = rcutils_get_zero_initialized_shared_library();
+  rcutils_ret_t ret = rcutils_load_shared_library(
+    &lib,
+    library_path.c_str(),
+    rcutils_get_default_allocator());
+  if (ret != RCUTILS_RET_OK) {
+    if (ret == RCUTILS_RET_BAD_ALLOC) {
+      throw std::runtime_error{"rcutils shared_library exception: failed to allocate memory"};
+    } else if (ret == RCUTILS_RET_INVALID_ARGUMENT) {
+      throw std::runtime_error{"rcutils shared_library exception: invalid arguments"};
```

Comment:
How can this be the case?

Reply:
Doesn't look like it can be, but it should still be checked no? What if there's a bug in the underlying implementation that causes this error?

Reply:
Yeah, checking it nevetheless is good. Maybe just add a comment that this shouldn't be possible.